### PR TITLE
Scoping changes and narrowing of injection selectors

### PR DESCRIPTION
**Pull request by infininight**
Reasoning left to individual commit messages, ask if there any questions.

### CODE REVIEW

File: Syntaxes/Markdown (GitHub Italics).tmLanguage [hunk 1]
```diff
@@ -2,12 +2,14 @@
...
 <dict>
+	<key>fileTypes</key>
+	<array/>
 	<key>hideFromUser</key>
 	<true/>
 	<key>injectionSelector</key>
 	<string>(L:text.html.markdown markup.italic.markdown)</string>
 	<key>name</key>
-	<string>Markdown (GitHub Injected)</string>
+	<string>Markdown (GitHub Italics)</string>
```
Comment (MikeMcQuaid) [inline Syntaxes/Markdown (GitHub Italics).tmLanguage:12]: @infininight Hmm, I'm not sure I agree with this rename as it's handling more than just italics. Maybe just `Markdown (GitHub)`?

**Comment by MikeMcQuaid** [comment]
@infininight My reasoning with the theming changes were that the Markdown bundle already makes similar e.g. heading-based theme changes but these are inconsistent with the output of GitHub's Markdown. I pulled just the font settings into https://github.com/mikemcquaid/GitHub-Markdown-Font-Settings.tmbundle; do all of these theme settings need pulled in there too? If so, would it be possible to have this also included in the TextMate 2 bundle list so that people who want to receive a more optimise theme can do so? Thanks for all the work here and review 🙇

**Comment by infininight** [comment]
Yeah for default bundles we want to provide some basic defaults like italics and bold, or in the case of the Diff bundle the colored lines for changed/removed. Generally though we want to avoid anything really changing unexpectedly. For instance I don't like for bundles to override the save command to run a validation command every save, or in this case installing a language grammar and the styling changing.

But it's all about expectations, a bundle exclusively for the theming would be fine. We have one in the installer that does basically the same for Latex.

BTW: As for the heading font/sizes, this is generally something much more radical than we would do as a default. It was mainly done to be a highlight of what is possible in 2.0 that wasn't possible in 1.x

File: Syntaxes/Markdown (GitHub Italics).tmLanguage [hunk 2]
```diff
@@ -3,20 +3,13 @@
...
 	<key>hideFromUser</key>
 	<true/>
 	<key>injectionSelector</key>
-	<string>(L:text.html.markdown)</string>
-	<key>keyEquivalent</key>
-	<string>^~M</string>
+	<string>(L:text.html.markdown markup.italic.markdown)</string>
 	<key>name</key>
-	<string>Markdown (GitHub Injected)</string>
+	<string>Markdown (GitHub Italics)</string>
```
Comment (MikeMcQuaid) [inline Syntaxes/Markdown (GitHub Italics).tmLanguage:12]: @infininight Still think this name should probably just be `Markdown (GitHub)` as it's doing more than just italics.

File: Syntaxes/Markdown (GitHub Italics).tmLanguage [hunk 3]
```diff
@@ -3,20 +3,13 @@
...
-		<string>markdn</string>
-	</array>
+	<array/>
 	<key>hideFromUser</key>
 	<true/>
 	<key>injectionSelector</key>
-	<string>(L:text.html.markdown)</string>
-	<key>keyEquivalent</key>
-	<string>^~M</string>
+	<string>(L:text.html.markdown markup.italic.markdown)</string>
```
Comment (MikeMcQuaid) [inline Syntaxes/Markdown (GitHub Italics).tmLanguage:10]: Not sure I understand the `markup.italic.markdown` bit?

Comment (MikeMcQuaid) [inline Syntaxes/Markdown (GitHub Italics).tmLanguage:10]: Oh, never mind, I see it now.

File: Syntaxes/Markdown (GitHub Italics).tmLanguage (hunk 2 above)
Comment (MikeMcQuaid) [inline Syntaxes/Markdown (GitHub Italics).tmLanguage:12]: Ignore me, I see it now 👍

**Comment by MikeMcQuaid** [comment]
Thanks for the PR @infininight! I've migrated the Preferences to https://github.com/mikemcquaid/GitHub-Markdown-Font-Settings.tmbundle. Let me know if there's anything else that I need to do that's blocking these being included in TextMate 2.

**Pull request merged**